kai/reactive_codeplanner/agent/analyzer_fix/agent.py

```diff
@@ -203,11 +214,13 @@ def guess_language(code: str, filename: Optional[str] = None) -> str:
         all_lexers = lexers.get_all_lexers()
         largest_rv_lexer = None
         largest_rv = 0.0
+        not_found_lexers: list[str] = []
         for name, _, _, _ in all_lexers:
             try:
                 lexer_found = cast(LexerMeta, lexers.get_lexer_by_name(name))
             except ClassNotFound:
-                logger.log(DEBUG, "unable to find lexer class for name: %s", name)
+                # FIXME: This is adding significant time to our request
```

Comment:
How significant?

Reply:
I put this test script together:

```python
import time

from kai.reactive_codeplanner.agent.analyzer_fix.agent import guess_language


time_before = time.time()

guess_language(
"""
import java.io.*;

public class Main {
    public static void main(String[] args) {
        System.out.println("Hello, World!");
    }
}
""",
"tester.java"
)

time_after = time.time()

print(f"Time taken: {time_after - time_before} seconds")
```

Here are the results:
```sh
(venv) [jonah@fedora kai-jonah]$ python tester.py
Time taken: 5.306198358535767 seconds
(venv) [jonah@fedora kai-jonah]$ python tester.py
Time taken: 4.978027820587158 seconds
(venv) [jonah@fedora kai-jonah]$ python tester.py
Time taken: 4.903670072555542 seconds
```

Reply:
wow yeah that's bad